packages/deployer/src/compiler.ts

```diff
@@ -207,6 +237,36 @@ export class Compiler {
         const sourceMapRuntime = compiled.contracts[contractIdentifier].srcmapRuntime;
         const sources = _.keys(compiled.sources);
         const updated_at = Date.now();
+
+        // There is no function overloading in typescript, so we must change the Typescript ABI interface.
+        // Overloaded function names are incremented as follows: functionName, functionName_2, functioname_3, ...
+        // If functionName_N already exists then compilation will fail.
+        const functionList:FunctionList = {};
+        for(let i = 0; i < abi.length; ++i) {
+            const type:string = abi[i].type;
+            if(type !== "function") continue;
+
+            // Construct possible function names
+            const originalName:string = (abi[i] as any).name;
+            const overloadedName:string = (originalName in functionList) ? originalName+"_"+(++functionList[originalName]) : originalName;
+            const unoverloadedName:string = originalName.indexOf("_") > 0 ? originalName.substr(0, originalName.lastIndexOf("_")) : originalName;
+
+            // Fail if an overloaded function was renamed to a function that already exists
+            if((overloadedName != originalName && overloadedName in functionList) ||
+                (unoverloadedName != originalName && unoverloadedName in functionList))
```

Comment:
This is a lot of conditionals to process. Let's assign them to an understandable variable name to reduce the cognitive burden on the reader:

```
const didOverloadedFunctionNameCreateCollision = (overloadedName != originalName && overloadedName in functionList) || (unoverloadedName != originalName && unoverloadedName in functionList);
if (didOverloadedFunctionNameCreateCollision) {
  ...
```

This way a casual reader knows **what** the conditional checks, and can dig into the nuts and bolts if he wants, but ins't forced to just to understand what the code is doing. 

It also removes the need for the code comment.